src/App.tsx

```diff
@@ -1,26 +1,118 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useState } from 'react';
+import { Todo } from './types/Todo';
+import { USER_ID, addTodo, deleteTodo, getTodos } from './api/todos';
+import { ErrorStatus } from './types/ErrorStatus';
+import { Status } from './types/Status';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoItem } from './components/TodoItem';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [error, setError] = useState<ErrorStatus>(ErrorStatus.Empty);
+  const [filter, setFilter] = useState<Status>(Status.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+
+  useEffect(() => {
+    getTodos()
+      .then(res => setTodoList(res))
+      .catch(() => {
+        setError(ErrorStatus.UnableToLoad);
+      });
+  }, []);
+
+  const filteredTodoList = (): Todo[] => {
+    switch (filter) {
+      case Status.Active:
+        return todoList.filter(todo => !todo.completed);
+      case Status.Completed:
+        return todoList.filter(todo => todo.completed);
+      default:
+        return todoList;
+    }
+  };
+
+  const onAddTodo = async (todoTitle: string) => {
+    setTempTodo({ id: 0, title: todoTitle, completed: false, userId: USER_ID });
+    try {
+      const newTodo = await addTodo({ title: todoTitle, completed: false });
+
+      setTodoList(prev => [...prev, newTodo]);
+    } catch (err) {
+      setError(ErrorStatus.UnableToAdd);
+      throw err;
+    } finally {
+      setTempTodo(null);
+    }
+  };
+
+  const onRemoveTodo = async (todoId: number) => {
```

Comment:
```suggestion
  const handleRemoveTodo = async (todoId: number) => {
```

https://medium.com/@master_43681/understanding-and-proper-usage-of-on-and-handle-prefixes-in-react-function-naming-119f0beea400